plugin/trace/trace.go

```diff
@@ -112,10 +116,9 @@ func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	span.SetTag(tagProto, req.Proto())
 	span.SetTag(tagRemote, req.IP())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
+	if t.EndpointType == "datadog" {
```

Comment:
Maybe put this type-specific logic up in the `OnStartup` by adding a function pointer to the `trace` struct and calling that here?

Reply:
Moved that up there